tutorials/add_model.md

```diff
@@ -0,0 +1,139 @@
+ 
+#### More on the score function 
+When you perform rating prediction evaluation (e.g., in terms of RMSE) Cornac calls the `Recommender.rate()` function, which in turn calls the `score()` function and clips the predicted values to lie in the same range as the original ratings. Analogously, Cornac calls the `Recommender.rank()` function, which returns a ranking of items according to their scores as predicted by the function `score()`. We recommend you to take a look at the `rate()` and `rank()` functions inside the `Recommender` class, as you may wish to redefine them as well, if the default settings do not meet your needs. 
+
+### 2. Making your recommender available to Cornac
+As you may have already noticed, Cornac treats each recommender model as a separate module, so as to reduce coupling and ease the contribution of new models. This is the reason why you need to include a `pmf/__init__.py` file:
+```python
+from .recom_pmf import PMF
+```
+
+One last step is required to make you recommender model available to Cornac, which is to update `../models/__init__.py` by adding the following line:
+```python
+from .pmf import PMF
+```
+
+### 3. Indicating dependencies
+The file `requirements.txt` is optional as opposed to the others, and it is only needed if your implementation relies on some external libraries/frameworks, e.g., TensorFlow, PyTorch, etc., in which case you should include this file and indicate which versions of the framework(s) are required. Here is a sample of a `requirements.txt` file:
+```
+tensorflow>=1.10.0
+```
+
+### 4. Adding documentation
+
+Cornac uses [Sphinx](http://www.sphinx-doc.org/en/master/) to generate documentation. All you need to do is to fulfill the Parameters section inside `recom_pmf.py`. If you are not familiar with Sphinx syntax conventions, you can just take a look at some of the existing Cornac models and follow them.
+
+Once you have documented the parameters of your recommender, the next step, to integrate your model to the docs, is to update `cornac/docs/source/models.rst` by adding the following code:
+```
+Probabilitic Matrix Factorization (PMF)
+-----------------------------------------
+.. automodule:: cornac.models.pmf.recom_pmf
+   :members:
+```
+
+At this point you are done, and ready to create a pull request, congratulation!
+
+### Including an example (optional)
+We highly encourage you to add an example on how to fit your model to some dataset and report the obtained results, see for instance `./cornac/examples/pcrl_example.py`. All examples should be added to `./cornac/examples/`. 
+
+
+## Summary
+
+In short, contributing a recommender model to Cornac involves,
+
+- Creating new files
+    - [x] ./cornac/models/model_name/\_\_init__.py
+    - [x] ./cornac/models/model_name/recom_model_name.py
+    - [x] ./cornac/models/model_name/requirement.py (optional)
+- Redefining two functions
```

Comment:
Implementing two functions

Reply:
Thanks for this suggestion, redefining seems more precise in this case?

Reply:
I think implement is more accurate because the functions are empty by default. There is nothing to redefine here.

Reply:
Ok, I am fine with both. Here redefine is to signal that these are inherited functions. I will change to implement.